server/metrics/measurement.go

```diff
@@ -69,27 +69,8 @@ func ClearMeasurementContext(ctx context.Context) context.Context {
 }
 
 func (m *Measurement) CountOkForScope(scope tally.Scope, tags map[string]string) {
-	switch scope {
-	case AuthOkCount:
-		if config.DefaultConfig.Metrics.Auth.Enabled {
-			m.countOk(scope, tags)
-		}
-	case RequestsOkCount:
-		if config.DefaultConfig.Metrics.Requests.Enabled && config.DefaultConfig.Metrics.Requests.Counter.OkEnabled {
-			m.countOk(scope, tags)
-		}
-	case FdbOkCount:
-		if config.DefaultConfig.Metrics.Fdb.Enabled && config.DefaultConfig.Metrics.Fdb.Counter.OkEnabled {
-			m.countOk(scope, tags)
-		}
-	case SessionOkCount:
-		if config.DefaultConfig.Metrics.Session.Enabled && config.DefaultConfig.Metrics.Session.Counter.OkEnabled {
-			m.countOk(scope, tags)
-		}
-	case SearchOkCount:
-		if config.DefaultConfig.Metrics.Search.Enabled && config.DefaultConfig.Metrics.Session.Counter.OkEnabled {
-			m.countOk(scope, tags)
-		}
+	if scope != nil {
```

Comment:
why do we need a scope nil check? Can we guarantee during initialization that if it is not enabled then we pass noopScope? Otherwise, if we miss this check at some place then this may happen again.